pygeo/constraints/baseConstraint.py

```diff
@@ -23,13 +24,15 @@ def __init__(self, name, nCon, lower, upper, scale, DVGeo, addToPyOpt):
         self.DVGeo = DVGeo
         self.addToPyOpt = addToPyOpt
 
-    def setDesignVars(self, x):
-        """
-        take in the design var vector from pyopt and set the variables for this constraint
-        This function is constraint specific, so the baseclass doesn't implement anything.
-        """
-        pass
+    # @abstractmethod
```

Comment:
I assume this is not use because we don't _always_ need this method? I don't get which criterion you used for abstract methods below, I am not very familiar with abstract classes.

Reply:
I don't actually fully understand what this is doing, I think it's for constraints that are analytically based on some DVs or something. In any case, this is not needed for all, so I have opted to not include it in the abstract class. Maybe I should just take this out.

Reply:
Yeah I think it's fine to just remove it instead of leave it commented